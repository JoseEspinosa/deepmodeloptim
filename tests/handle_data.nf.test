
nextflow_workflow {

    name "Test Workflow HANDLE_DATA"
    script "../workflows/handle_data.nf"
    workflow "HANDLE_DATA"

    test("Test reproducibility of data handling") {

        when {

            // TODO Check that continues working with $NXF_SINGULARITY_CACHEDIR
            // // make so that nf-test will see a singularity cache dir where is usally set to, again interpretation of projectdir has to be done here and not rely on an external config.
            // params {
            //     singularity_cache_dir = "${projectDir}/singularity_cache"
            // }

            // repeating in input[0-1] the same content of params.csv and parma.exp_conf present in the test.config because nf-test does not interpret the ${projectDir} correctly otherwise
            workflow {
                """
                input[0] = "${projectDir}/bin/tests/test_data/dna_experiment/test_with_split.csv"
                input[1] = "${projectDir}/examples/test.json"
                input[2] = Channel.of("TESTING")
                """
            }
        }

        then {

            assert workflow.success

            assert snapshot(workflow.out.data).match()
        }
    }
}
