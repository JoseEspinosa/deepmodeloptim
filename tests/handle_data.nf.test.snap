{
    "Test reproducibility of data handling": {
        "content": [
            [
                [
                    "test_with_split.csv - shuffle",
                    "no_split",
                    "test_with_split-shuffled-experiment.json:md5,04768aaf767c8cd2c043f510432b78e0",
                    "test_with_split-shuffle.csv:md5,920c64271a3bd74fa1ab9015fcc8f905"
                ],
                [
                    "test_with_split.csv - test-1",
                    "no_split",
                    "test-1-experiment.json:md5,4ce80948f2bd82af2ca42e3bb81fbb32",
                    "test_with_split-test-1-trans.csv:md5,8dbc69f2845da31ff8ab2381a2667a4e"
                ],
                [
                    "test_with_split.csv - test-2",
                    "RandomSplitter_0.34_0.33_0.33_4",
                    "test-2-experiment.json:md5,5bb998041e683c327dcfec81a908fa36",
                    "test_with_split-test-2-trans.csv:md5,64ecd933e2bc0c0d2c42a107288c3a0e"
                ],
                [
                    "test_with_split.csv - test-3",
                    "no_split",
                    "test-3-experiment.json:md5,2a6576e2eee24cbd026e3db7944d2733",
                    "test_with_split-test-3-trans.csv:md5,9f0179b00db4bd806df0d0760e036020"
                ],
                [
                    "test_with_split.csv - test-4",
                    "RandomSplitter_0.34_0.33_0.33_4",
                    "test-4-experiment.json:md5,888b5a8cdf10c3f6fbd0c061c127b28b",
                    "test_with_split-test-4-trans.csv:md5,35c600e3d5af5280c3ce50e683d4e1da"
                ],
                [
                    "test_with_split.csv - test-5",
                    "no_split",
                    "test-5-experiment.json:md5,0dd70a06361d79438e1cd154cacf132b",
                    "test_with_split-test-5-trans.csv:md5,535ede6479723133ff71020eec9de4b1"
                ],
                [
                    "test_with_split.csv - test-6",
                    "RandomSplitter_0.34_0.33_0.33_4",
                    "test-6-experiment.json:md5,fde920c7e4d815a56e4825dfd3f7e415",
                    "test_with_split-test-6-trans.csv:md5,fbbdfd06c6ec07b126ae68592bd09e54"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-07T17:10:32.646115"
    }
}